Formula/libsigc++.rb

```diff
@@ -53,7 +53,7 @@ def install
     EOS
 
     on_linux do
-      ENV["CXX"] = Formula["gcc"].opt_bin/"c++-10"
+      ENV["CXX"] = Formula["gcc"].opt_bin/"c++-#{Formula["gcc"].any_installed_version.major}"
     end
```

Comment:
If you remove this the correct compiler should already be chosen in the test block since you have `fails_with`.